### ISSUE TITLE
Replies to excluded comments should be excluded as well

### ISSUE THREAD

**Issue by drillprop** [Joystream/orion#93]
As the title says,`isExcluded` flag should be propagated to all replies

**Comment by Lezek123** [comment]
This is actually quite complex issue, because even if we solve this on database view level and correctly re-calculate `video.commentsCount`, `comment.repliesCount` etc., users can still add replies to an excluded comment, so there is this very hard to solve situation where:
1. Operator excludes a comment
2. User adds a deeply nested reply to the excluded comment (let's say, a reply to a reply to a reply to a reply... to the excluded comment)
3. How will the processor know it should not increase `video.commentsCount` in this case without recursive iteration over parents?

**Comment by Lezek123** [comment]
An alternative is to perhaps propagate the flag to child comments only when excluding content, but not do this when restoring excluded content (to avoid accidently restoring repilies than may have been separately excluded earlier) + whenever a comment is exluded, any reply to it also becomes automatically excluded when created.

**Comment by Lezek123** [comment]
Actually all the solutions I can think of have some major flaw, ie.:
- it's hard to avoid introducing some unconstrained recursion when propagating the status, which could be then maliciously triggered by the user,
- if we allow `video.commentsCount` to be updated both by the processor and by the graphql server, we may run into conflicts if both those processes try to do the update at more or less the same time,
- we also may run into conflicts if the processor already scheduled an insertion of a new reply to a comment we just excluded

I can propose a following approach as an alternative:
1. Excluding a comment doesn't actually remove it from the query results, but it still sets `isExcluded` flag to `true`. The content of the excluded comment (and the content inside related `CommentCreated`/`CommentTextUpdated` events) is overriden by a database view to something like `[CENSORED]`. Atlas can however still differentiate between actually censored comments and _fake censored comments_ (those that just have `[CENSORED]` as text) by inspecting the value of `isExcluded` flag if needed.
2. I leave it up to you to decide whether to count such comment or not in `video.commentsCount` / `parentComment.repliesCount`.

The key of this solution is that:
- The status is not propagated
- You can differentiate replies to excluded comments from the comments that aren't replies at all
- Comment content is hidden from public view, even when interacting directly w/ the api
- You can choose how to display the excluded comments in the UI or whether to display them at all

**Comment by Lezek123** [comment]
> Just to confirm - Isn't this problematic to do? I mean, you'll exclude from the results videos and channels but comments not? Won't this be hard to implement?

No, it's not a problem to implement this at all, I can specify any custom conditions when creating the database views.

> Let's just put here an empty string, maybe? The same as we have in deleted and moderated comments?

Sounds good

> Again, similar to moderated and deleted comments - they shouldn't be taken into account in repliesCount and commentCount IMO

Ok :+1: 

> The operator will still be able to see the comment text if he provides the operator key in the header, am I'm right?

Yes

**Comment by Lezek123** [comment]
Solution implemented in https://github.com/Joystream/orion/pull/49/commits/e47df82320bfd2b7e0d5acee93597eded8f004cb
To be fully consistent with `DELETED` / `MODERATED` comments and also to save some implementation time, I didn't replace the `text` in comment created/edited events, only in the comment itself.
